### PR TITLE
Meta data for template pages

### PR DESCRIPTION
**Pull request by victoriabernard92**
Have template pages collect title and description from the registry so search results don't look like:

image

Related https://github.com/cloudflare/workers-docs/issues/385

### CODE REVIEW

File: workers-site/src/main.js [hunk 1]
```diff
@@ -51,7 +51,19 @@ export async function handleRequest(event) {
       console.log('Handling redirect')
       return handleRedirect(request)
     }
-
+    // TODO remove this and just set the meta title/descriptions in Gatsby
+    if (pathname.includes('templates/pages')) {
+      //Grab the template's title from the registry
+      const templateId = pathname.replace(/.*pages\//, '')
+      const templateResp = await fetch(templateRegURL + templateId)
+      const templateJSON = await templateResp.json()
```
Comment (signalnerve) [inline workers-site/src/main.js]: love this solution, it's super rad 😍

File: workers-site/src/main.js [hunk 2]
```diff
@@ -51,7 +51,19 @@ export async function handleRequest(event) {
...
+      const metaInfo = {
+        title: templateJSON.title + ' - Cloudflare Workers Docs',
+        description: templateJSON.description,
+      }
+      // Rewrite all meta titles/descriptions the the correct title
+      return await new HTMLRewriter().on('meta', new MetaHandler(metaInfo)).transform(body)
```
Comment (signalnerve) [inline workers-site/src/main.js]: should we be rewriting the `title` tag too?

Comment (victoriabernard92) [reply workers-site/src/main.js]: Since we only get one HTMLRewriter that will be hard :(

Comment (signalnerve) [reply workers-site/src/main.js]: can we attach another handler on here?

```js
return await new HTMLRewriter()
  .on('meta', new MetaHandler(metaInfo))
  .on('title', new TitleHandler(metaInfo.title))
  .transform(body) 
```

Comment (victoriabernard92) [reply workers-site/src/main.js]: only the first transform handler will execute

Comment (signalnerve) [reply workers-site/src/main.js]: hey @victoriabernard92, i'm not sure i understand! you should be able to add multiple `.on` handlers, afaik. did this not work?

Comment (victoriabernard92) [reply workers-site/src/main.js]: Fixed :)

**Pull request merged**